module_utils/kafka_manager.py

```diff
@@ -827,11 +662,254 @@ def update_topic_assignment(self, json_assignment, zknode):
+                "client_id": "kafka-python-2.0.1",
+                "member_assignment": {
+                    "assignment": {
+                        "test_1605127023": [
+                            0
+                        ]
+                    },
+                    "user_data": "",
+                    "version": 0
+                },
+                "member_metadata": {
+                    "subscription": [
+                        "test_1605127023"
+                    ],
+                    "user_data": "",
+                    "version": 0
+                }
+            }
+        }
+    }
+}
+        """
+        consumer_groups = {}
+        for broker in self.get_brokers():
+            request = ListGroupsRequest_v2()
+            response = self.send_request_and_get_response(
+                request, node_id=broker.nodeId
+            )
+            if response.error_code != self.SUCCESS_CODE:
+                raise KafkaManagerError(
+                    'Error while list consumer groups of %s. '
+                    'Error key is %s, %s.' % (
+                        broker.nodeId,
+                        kafka.errors.for_code(response.error_code).message,
+                        kafka.errors.for_code(response.error_code).description
+                    )
+                )
+            if response.groups:
+                request = DescribeGroupsRequest_v0(
+                    groups=tuple(
+                        [group for group, protocol in response.groups]
+                    )
+                )
+                response = self.send_request_and_get_response(
+                    request, node_id=broker.nodeId
+                )
+                for err, gid, gstate, _, _, _members in response.groups:
+                    members = {}
+                    for mid, cid, chost, mdata, assign in _members:
+                        mdata = ProtocolMetadata.decode(mdata)
+                        assign = MemberAssignment.decode(assign)
+                        assignment = {}
+                        for t, p in assign.assignment:
+                            assignment[t] = p
+                        members[mid] = {
+                            'client_id': cid,
+                            'client_host': chost,
+                            'member_metadata': {
+                                'version': mdata.version,
+                                'subscription': mdata.subscription,
+                                'user_data': mdata.user_data.decode('utf-8')
+                            },
+                            'member_assignment': {
+                                'version': assign.version,
+                                'assignment': assignment,
+                                'user_data': assign.user_data.decode('utf-8')
+                            }
+                        }
+                    group = {
+                        'error_code': err,
```

Comment:
May add `protocolType`.

Reply:
Thanks! Added `protocol_type` and `protocol`, because I did not want to make any jealous :)